lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -274,8 +274,11 @@ sub _schedule_iso {
                 my $job = $jobs_resultset->create_from_settings($settings, $self->id);
                 push @created_jobs, $job;
 
-                $testsuite_ids{$settings->{TEST_SUITE_NAME}}->{$settings->{MACHINE}} //= [];
-                push @{$testsuite_ids{$settings->{TEST_SUITE_NAME}}->{$settings->{MACHINE}}}, $job->id;
+                $testsuite_ids{JOB_TEMPLATE_NAME}->{$settings->{TEST}}->{$settings->{MACHINE}} //= [];
```

Comment:
The name of this variable was misleading before - it should have been `%job_ids_by_machine_by_test_suite_name`. Not it should be  `%job_ids_by_machine_by_test_suite_name_by_origin_of_settings`. Of course it shouldn't be called this way. My real point is that this becomes too complicated. Instead, this might be a place to unify the settings from the different sources. This would help to avoid the repetitive code patterns you introduce in subsequent changes to to read settings preferably from job template settings.

Reply:
Actually the job template name is what we care about. If it's not defined, the test suite name is the job template name... there is no case where both are used at the same time